File: patched-dex/src/main/java/com/samsung/android/shealthmonitor/ecg/util/DialogUtils.java
package com.samsung.android.shealthmonitor.ecg.util;

import android.content.Intent;
import android.content.pm.PackageManager;
import android.content.res.Resources;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentActivity;
import com.samsung.android.shealthmonitor.bp.R;
import com.samsung.android.shealthmonitor.ui.dialog.SAlertDlgFragment;
import com.samsung.android.shealthmonitor.ui.dialog.listener.OnPositiveButtonClickListener;
import com.samsung.android.shealthmonitor.util.LOG;
import kotlin.jvm.internal.Intrinsics;

/* compiled from: DialogUtils.kt */
public final class DialogUtils {
    public static final DialogUtils INSTANCE = new DialogUtils();
    private static final String NO_CONNECTION_POPUP_TAG = "NO_CONNECTION_POPUP_TAG";
    private static final String TAG = "S HealthMonitor - DialogUtils";

    private DialogUtils() {
    }

    public final String getTAG() {
        return TAG;
    }

    public final String getNO_CONNECTION_POPUP_TAG() {
        return NO_CONNECTION_POPUP_TAG;
    }

    public static /* synthetic */ void showNoConnectionDialog$default(DialogUtils dialogUtils, FragmentActivity fragmentActivity, OnPositiveButtonClickListener onPositiveButtonClickListener, int i, Object obj) {
        if ((i & 2) != 0) {
            onPositiveButtonClickListener = null;
        }
        dialogUtils.showNoConnectionDialog(fragmentActivity, onPositiveButtonClickListener);
    }

    public final void showNoConnectionDialog(FragmentActivity fragmentActivity, OnPositiveButtonClickListener onPositiveButtonClickListener) {
        Intrinsics.checkParameterIsNotNull(fragmentActivity, "activity");
        if (!fragmentActivity.isFinishing() && !fragmentActivity.isDestroyed()) {
            SAlertDlgFragment.Builder builder = new SAlertDlgFragment.Builder(R.string.cannot_connect_with_your_watch, 1);
            builder.setContentText(R.string.cannot_connect_with_your_watch_desc);
            Fragment findFragmentByTag = fragmentActivity.getSupportFragmentManager().findFragmentByTag(NO_CONNECTION_POPUP_TAG);
            if (findFragmentByTag != null) {
                ((SAlertDlgFragment) findFragmentByTag).dismiss();
            }
            builder.setPositiveButtonClickListener(R.string.baseui_button_ok, new DialogUtils$showNoConnectionDialog$1(fragmentActivity, onPositiveButtonClickListener));
            builder.setPositiveButtonTextColor(fragmentActivity.getResources().getColor(R.color.dialog_color, (Resources.Theme) null));
            builder.setCanceledOnTouchOutside(false);
            try {
                builder.build().show(fragmentActivity.getSupportFragmentManager(), NO_CONNECTION_POPUP_TAG);
                String str = TAG;
                LOG.d(str, "Show " + NO_CONNECTION_POPUP_TAG);
            } catch (Exception unused) {
                String str2 = TAG;
                LOG.d(str2, "Error in " + NO_CONNECTION_POPUP_TAG);
            }
        }
    }

    public final void launchWearableManager(FragmentActivity fragmentActivity) {
        Intrinsics.checkParameterIsNotNull(fragmentActivity, "activity");
        PackageManager packageManager = fragmentActivity.getPackageManager();
        try {
            if (packageManager.getApplicationInfo("com.samsung.android.app.watchmanager", 0).enabled) {
                Intent launchIntentForPackage = packageManager.getLaunchIntentForPackage("com.samsung.android.app.watchmanager");
                if (launchIntentForPackage == null) {
                    LOG.i(TAG, "launchWearableManager() : intent is null.");
                    return;
                }
                launchIntentForPackage.setFlags(268468224);
                fragmentActivity.startActivity(launchIntentForPackage);
                return;
            }
            String str = TAG;
            LOG.i(str, "launchWearableManager() : " + fragmentActivity.getPackageName() + " is not enabled.");
        } catch (PackageManager.NameNotFoundException unused) {
            LOG.e(TAG, "launchWearableManager() : NameNotFoundException is occurred");
        }
    }

    public final void removeFragmentDialog(FragmentActivity fragmentActivity, String... strArr) {
        Intrinsics.checkParameterIsNotNull(fragmentActivity, "activity");
        Intrinsics.checkParameterIsNotNull(strArr, "tags");
        if (!fragmentActivity.isFinishing() && !fragmentActivity.isDestroyed()) {
            for (String findFragmentByTag : strArr) {
                Fragment findFragmentByTag2 = fragmentActivity.getSupportFragmentManager().findFragmentByTag(findFragmentByTag);
                if (findFragmentByTag2 != null) {
                    ((SAlertDlgFragment) findFragmentByTag2).dismiss();
                }
            }
        }
    }
}

File: patched-dex/shm161_mymod_mod/unknown/com/samsung/android/shealthmonitor/bp/util/DialogUtils.kt
package com.samsung.android.shealthmonitor.bp.util

import android.content.Intent
import android.content.pm.PackageManager
import android.support.v4.app.FragmentActivity
import com.samsung.android.shealthmonitor.bp.R
import com.samsung.android.shealthmonitor.ui.dialog.SAlertDlgFragment
import com.samsung.android.shealthmonitor.ui.dialog.listener.OnPositiveButtonClickListener
import com.samsung.android.shealthmonitor.util.LOG


object DialogUtils {

    val TAG = LOG.prefix + "DialogUtils"
    val NO_CONNECTION_POPUP_TAG = "NO_CONNECTION_POPUP_TAG"

    fun showNoConnectionDialog(activity : FragmentActivity, okListener : OnPositiveButtonClickListener? = null) {
        if (!activity.isFinishing && !activity.isDestroyed) {

            val builder = SAlertDlgFragment.Builder(R.string.cannot_connect_with_your_watch,
                    SAlertDlgFragment.TYPE_P_BUTTON)
            builder.setContentText(R.string.cannot_connect_with_your_watch_desc)

            val found = activity.supportFragmentManager.findFragmentByTag(NO_CONNECTION_POPUP_TAG)
            if (found !=null) {
                (found as SAlertDlgFragment).dismiss()
            }

            builder.setPositiveButtonClickListener(R.string.baseui_button_ok) {
                launchWearableManager(activity)
                if (okListener != null) {
                    okListener.onClick(it)
                }
            }

            builder.setPositiveButtonTextColor(activity.resources.getColor(R.color.dialog_color, null))
            builder.setCanceledOnTouchOutside(false)
            try {
                builder.build().show(activity.supportFragmentManager, NO_CONNECTION_POPUP_TAG)
                LOG.d(TAG, "Show $NO_CONNECTION_POPUP_TAG")
            } catch (e: Exception) {
                LOG.d(TAG, "Error in $NO_CONNECTION_POPUP_TAG")
            }
        }
    }

    fun launchWearableManager(activity : FragmentActivity) {
        val manager = activity.packageManager
        val wearableManagerPackage = "com.samsung.android.app.watchmanager"
        try {
            val info = manager.getApplicationInfo(wearableManagerPackage, 0)
            if (info.enabled) {
                val intent = manager.getLaunchIntentForPackage(wearableManagerPackage)
                if (intent == null) {
                    LOG.i(TAG, "launchWearableManager() : intent is null.")
                    return
                }
                intent.flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK
                activity.startActivity(intent)
            } else {
                LOG.i(TAG, "launchWearableManager() : ${activity.packageName} is not enabled.")
            }
        } catch (e: PackageManager.NameNotFoundException) {
            LOG.e(TAG, "launchWearableManager() : NameNotFoundException is occurred")
        }
    }

    fun removeFragmentDialog(activity : FragmentActivity, vararg tags : String ) {
        if (!activity.isFinishing && !activity.isDestroyed) {
            for (tag in tags) {
                val found = activity.supportFragmentManager.findFragmentByTag(tag)
                if (found !=null) {
                    (found as SAlertDlgFragment).dismiss()
                }
            }
        }
    }
}